runtime/vm/xcinterp.m4

```diff
@@ -165,7 +165,11 @@ START_PROC(c_cInterpreter)
 	mov uword ptr J9TR_ELS_jitFPRegisterStorageBase[_rax],_rbx
 C_FUNCTION_SYMBOL(cInterpreter):
 	mov _rax,uword ptr J9TR_VMThread_javaVM[_rbp]
+ifdef({OSX},{
```

Comment:
What's the problem here? I'd rather not see platform ifdefs in the common code. Can the macro be fixed?

Reply:
The problem is that on OSX, `CALL_C_WITH_VMTHREAD(uword ptr J9TR_JavaVM_bytecodeLoop[_rax],0)` gets processed as `call _qword ptr 192[rax]`. The underscore prepended to `qword` causes an error. This is a result of `CALL_C_WITH_VMTHREAD` making use of the `CALL_C_FUNC` macro, which then makes use of the `C_FUNCTION_SYMBOL` macro. On OSX:  `define({C_FUNCTION_SYMBOL},_$1)`. It is meant to handle symbol names, as opposed to addresses. It will work on Windows and Linux since they do not need to modify the symbol names, but it will not work on Mac. I will get rid of the platform ifdef in common code problem by introducing a different macro for making such calls, and update this branch.